LazyCache/CachingService.cs

```diff
@@ -92,7 +92,16 @@ public virtual bool TryGetValue<T>(string key, out T value)
         {
             ValidateKey(key);
 
-            return CacheProvider.TryGetValue(key, out value);
+            try
+            {
+                var contains = CacheProvider.TryGetValue<Lazy<T>>(key, out var lazyFactory);

```

Comment:
Could you use TryGetValue<object> and then do something to coerce the value, instead of relying on catching cast exceptions?  Also, as a larger issue, instead of storing both Lazy and non-Lazy values in CacheProvider, what if you just always stored a Lazy?  That way you don't have to figure out what is in there when retrieving the values.